app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -85,6 +84,223 @@ def registry_authorize
       return false
     end
 
+    def handle_blob_push(props = nil)
+      # Handle validation and repo creation for container pushes before talking to pulp
+      return false unless confirm_push_settings
+      props = parse_blob_push_props if props.nil?
+      return false unless check_blob_push_field_syntax(props)
+
+      # validate input and find the org and product either using downcase label or id
+      if props[:schema] == "label"
+        return false unless check_blob_push_org_label(props)
+        return false unless check_blob_push_product_label(props)
+      else
+        return false unless check_blob_push_org_id(props)
+        return false unless check_blob_push_product_id(props)
+      end
+
+      return false unless check_blob_push_container(props)
+      create_container_repo_if_needed
+      return true
+    end
+
+    def parse_blob_push_props(path_string = nil)
+      # path string should follow one of these formats:
+      #   - /v2/{org_label}/{product_label}/{name}/blobs/uploads...
+      #   - /v2/id/{org_id}/{product_id}/{name}/blobs/uploads...
+      #   - /v2/{org_label}/{product_label}/{name}/manifests/...
+      #   - /v2/id/{org_id}/{product_id}/{name}/manifests/...
+      # inputs not matching format will return {valid_format: false}
+      path_string = @_request.fullpath if path_string.nil?
+      segments = path_string.split('/')
+
+      if segments.length >= 7 && segments[0] == "" && segments[1] == "v2" &&
+        segments[2] != "id" && (segments[5] == "blobs" || segments[5] == "manifests")
+
+        return {
+          valid_format: true,
+          schema: "label",
+          organization: segments[2],
+          product: segments[3],
+          name: segments[4]
+        }
+      elsif segments.length >= 8 && segments[0] == "" && segments[1] == "v2" &&
+        segments[2] == "id" && (segments[6] == "blobs" || segments[6] == "manifests")
+
+        return {
+          valid_format: true,
+          schema: "id",
+          organization: segments[3],
+          product: segments[4],
+          name: segments[5]
+        }
+      else
+        return {valid_format: false}
+      end
+    end
+
+    def check_blob_push_field_syntax(props)
+      # check basic url field syntax
+      unless props[:valid_format]
+        return render_podman_error(
```

Comment:
The return might not be the issue, but it's one possibility.

Reply:
Ah, it says "Also note that neither redirect nor render terminate execution of the action"

So perhaps then there is a return missing that is causing a double render?

Reply:
It might have to do with how you're trying to render stuff in an after_action.

In the action, there should already be a render or a redirect, so there's no more API response to send inside after_action.

Reply:
""after" filters are registered via [after_action](https://api.rubyonrails.org/v7.1.3.3/classes/AbstractController/Callbacks/ClassMethods.html#method-i-after_action). They are similar to "before" filters, but because the action has already been run they have access to the response data that's about to be sent to the client. Obviously, "after" filters cannot stop the action from running. Please note that "after" filters are executed only after a successful action, but not when an exception is raised in the request cycle."

https://guides.rubyonrails.org/action_controller_overview.html#after-filters-and-around-filters

Reply:
Fixed, thank you. After discussion, we determined that content indexing only needs to run after manifest upload. after_action has been removed and this method is called from the manifest push.